src/QirRuntime/lib/QIR/intrinsicsMath.cpp

```diff
@@ -28,4 +39,38 @@ double quantum__qis__arctan2__body(double y, double x)
     return std::atan2(y, x);        // https://en.cppreference.com/w/cpp/numeric/math/atan2
 }
 
+int64_t quantum__qis__drawrandomint__body(int64_t minimum, int64_t maximum)
+{
+    if(minimum > maximum)
+    {
+        quantum__rt__fail(quantum__rt__string_create(Quantum::Qis::Internal::excStrDrawRandomInt));
+    }
+
+    // https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution
+    // https://en.cppreference.com/w/cpp/numeric/random
+    thread_local static std::mt19937_64 gen(seedIsRandom
+                                                ? std::random_device()() :  // Default
+                                                0);                         // For test purposes only.
+
+    return std::uniform_int_distribution<int64_t>(minimum, maximum)(gen);
+}
+
 }   // extern "C"
+
+namespace   // Visible in this translation unit only.
+{
+    thread_local bool seedIsRandom = true;
+}
+
+// For test purposes only:
+namespace Quantum { namespace Qis { namespace Internal 
```

Comment:
Please run clang formatter to have same line formatting across the lines (I know, you prefer more compact namespace declarations and we probably can change the formatter to do this rather than one namespace per line, but for now the automatically applied formatting will split the lines)

Reply:
Fixed.